vernac/declaremods.ml

```diff
@@ -468,13 +468,8 @@ and collect_export f (f',mp) (exports,objs as acc) =
         | Some f0 -> Some (filter_or f f0))
         exports
     in
-    (* If the map doesn't change there is nothing new to export.
-
-       It's possible that [filter_and] or [filter_or] mangled precise
-       filters such that we repeat uselessly, but the important
-       [Unfiltered] case is handled correctly.
-    *)
-    if exports == exports' then acc
+    (* If the map doesn't change there is nothing new to export. *)
+    if MPmap.equal filter_eq exports exports' then acc
```

Comment:
You should still use physical equality as a fast-path, crawling over the map may be inefficient.

Reply:
Changed to preserve physical equality in the `update` using a filter_eq check.